PowerEditor/src/NppCommands.cpp

```diff
@@ -682,18 +683,34 @@ void Notepad_plus::command(int id)
 		break;
 
 		case IDM_VIEW_PROJECT_PANEL_1:
-		{
-			launchProjectPanel(id, &_pProjectPanel_1, 0);
-		}
-		break;
 		case IDM_VIEW_PROJECT_PANEL_2:
-		{
-			launchProjectPanel(id, &_pProjectPanel_2, 1);
-		}
-		break;
 		case IDM_VIEW_PROJECT_PANEL_3:
 		{
-			launchProjectPanel(id, &_pProjectPanel_3, 2);
+			ProjectPanel** pp [] = {&_pProjectPanel_1, &_pProjectPanel_2, &_pProjectPanel_3};
+			int idx = id - IDM_VIEW_PROJECT_PANEL_1;
+			if (*pp [idx] == nullptr)
+			{
+				launchProjectPanel(id, pp [idx], idx);
+			}
+			else
+			{
+				if (not (*pp[idx])->isClosed())

```

Comment:
Next time, please use modern operator (`!` instead of `not`).

Reply:
I tend to like _not_ myself, but if you want ! then now I know it.